integration-cli/docker_cli_rmi_test.go

```diff
@@ -268,3 +268,33 @@ func (s *DockerSuite) TestRmiBlank(c *check.C) {
 		c.Fatalf("Expected error message not generated: %s", out)
 	}
 }
+
+func (s *DockerSuite) TestRmiContainerImageNotFound(c *check.C) {
+	// Build 2 image for testing.
+	imageNames := []string{"test1", "test2"}
+	imageIds := make([]string, 2)
+	for i, name := range imageNames {
+		dockerfile := fmt.Sprintf("FROM busybox\nMAINTAINER %s\nRUN echo %s\n", name, name)
+		id, err := buildImage(name, dockerfile, false)
+		c.Assert(err, check.IsNil)
+		imageIds[i] = id
+	}
+
+	// Create a long-running container.
+	_, status := dockerCmd(c, "run", "-d", imageNames[0], "top")
```

Comment:
Status will always be zero here because `dockerCmd` will fail the test immediately otherwise.